core/trino-main/src/main/java/io/trino/execution/QueryManagerConfig.java

```diff
@@ -52,10 +53,10 @@
 })
 public class QueryManagerConfig
 {
-    public static final long AVAILABLE_HEAP_MEMORY = Runtime.getRuntime().maxMemory();
+    public static final long AVAILABLE_HEAP_MEMORY = getRuntime().maxMemory();
     public static final int MAX_TASK_RETRY_ATTEMPTS = 126;
     public static final int FAULT_TOLERANT_EXECUTION_MAX_PARTITION_COUNT_LIMIT = 1000;
-
+    public static final int DISPATCHER_THREADPOOL_MAX_SIZE = getRuntime().availableProcessors() * 10;
```

Comment:
why is this specific default chosen? - please leave a code comment.

Can it lead to regressions for people when updating?

Reply:
> why is this specific default chosen? - please leave a code comment.
> 
> Can it lead to regressions for people when updating?

thanks @hashhar for reviewing!

My intention was to provide a config, so that we will have leverage to control it.
You are correct, this might be a smaller number, so may be we can keep it to a reasonable big number like `10,000` , and if anyone sees an issue can reduce it.

Reply:
Based on the discussion, how about we go with
```
public static final int DISPATCHER_THREADPOOL_MAX_SIZE = max(16, getRuntime().availableProcessors() * 2);
```
This is just a limit the actual number isn't a huge deal if it is larger, but it could be a problem is someone is running on a 2 or 4 core machine.